pytest_fixtures/component/host.py

```diff
@@ -52,13 +52,24 @@ def setup_rhst_repo():
     return {'ak': ak, 'cv': cv, 'lce': lce, 'org': org, 'repo_name': repo_name}
 
 
+@pytest.fixture(scope='module')
+def module_host_template(module_org, smart_proxy_location, module_target_sat):
+    host_template = module_target_sat.api.Host(
+        organization=module_org, location=smart_proxy_location, build=True
+    )
+    host_template.create_missing()
+    host_template.name = None
+    return host_template
+
+
 @pytest.fixture
 def host_ui_options(module_host_template):
     os_name = (
         f'{module_host_template.operatingsystem.name} {module_host_template.operatingsystem.major}'
     )
+    name = gen_string('alpha').lower()
     values = {
-        'host.name': module_host_template.name,
+        'host.name': name,
         'host.organization': module_host_template.organization.name,
```

Comment:
Is there a specific reason why you have to set `host_template.name = None` and then generate a random one in `host_ui_options`? Would it make more sense to generate the random name in `module_host_template` so that you don't have to generate a name every time you use it?

Reply:
so the module_host_template is actually generating a random name, issue here is that it is module scoped, so the name would be the same for multiple hosts.